### PR TITLE
Refactoring somethings

### PR DESCRIPTION
**Pull request by 0x1306e6d**
Refactoring Beans
Refactoring Tokenizer
Extract unicodes to CONSTANT

### CODE REVIEW

File: src/com/esotericsoftware/yamlbeans/Beans.java [hunk 1]
```diff
@@ -38,253 +38,370 @@
...
+ * Utility for dealing with beans and public fields.
+ *
+ * @author <a href="mailto:misc@n4te.com">Nathan Sweet</a>
+ */
 class Beans {
-	private Beans () {
+
+	static private final String GETTER = "get";
+	static private final String GETTER_BOOLEAN = "is";
+	static private final String SETTER = "set";
```
Comment (NathanSweet) [inline src/com/esotericsoftware/yamlbeans/Beans.java]: I'm not a fan of using constants for such strings.

File: src/com/esotericsoftware/yamlbeans/Beans.java [hunk 2]
```diff
@@ -38,253 +38,370 @@
...
+		}
 		try {
 			Class constructorProperties = Class.forName("java.beans.ConstructorProperties");
+
 			for (Constructor typeConstructor : type.getConstructors()) {
 				Annotation annotation = typeConstructor.getAnnotation(constructorProperties);
-				if (annotation == null) continue;
-				String[] parameterNames = (String[])constructorProperties.getMethod("value").invoke(annotation, (Object[])null);
-				return new DeferredConstruction(typeConstructor, parameterNames);
+				if (annotation == null) {
```
Comment (NathanSweet) [inline src/com/esotericsoftware/yamlbeans/Beans.java]: There is no reason to add curly braces. Please follow the style the project is already using.

File: src/com/esotericsoftware/yamlbeans/Beans.java [hunk 3]
```diff
@@ -38,253 +38,370 @@
...
-		return null;
+
+		return commonImplementationConstructor;
 	}
 
-	static private ArrayList<Field> getAllFields (Class type) {
-		ArrayList<Class> classes = new ArrayList();
+	static private ArrayList<Field> getAllFields(Class type) {
+		ArrayList<Class> classes = new ArrayList<Class>();
+		ArrayList<Field> allFields = new ArrayList<Field>();
```
Comment (NathanSweet) [inline src/com/esotericsoftware/yamlbeans/Beans.java]: Moving all declarations first hurts readability will not be merged. Are you using automated tools to make these changes?

File: src/com/esotericsoftware/yamlbeans/Beans.java [hunk 4]
```diff
@@ -38,253 +38,370 @@
...
 			return getMethod.invoke(object);
 		}
 	}
 
 	static public class FieldProperty extends Property {
+
 		private final Field field;
 
-		public FieldProperty (Field field) {
+		public FieldProperty(Field field) {
```
Comment (NathanSweet) [inline src/com/esotericsoftware/yamlbeans/Beans.java]: Formatting for PRs should follow the project's style. This Eclipse formatter can be used:
https://github.com/libgdx/libgdx/blob/master/eclipse-formatter.xml

File: src/com/esotericsoftware/yamlbeans/emitter/EmitterWriter.java [hunk 5]
```diff
@@ -16,29 +16,34 @@
...
+		ESCAPE_REPLACEMENTS.put((int) Unicode.HORIZONTAL_TABULATION, "t");
+		ESCAPE_REPLACEMENTS.put((int) '\n', "n");
+		ESCAPE_REPLACEMENTS.put((int) Unicode.VERTICAL_TABULATION, "v");
+		ESCAPE_REPLACEMENTS.put((int) Unicode.FORM_FEED, "f");
+		ESCAPE_REPLACEMENTS.put((int) '\r', "r");
+		ESCAPE_REPLACEMENTS.put((int) Unicode.ESCAPE, "e");
+		ESCAPE_REPLACEMENTS.put((int) '"', "\"");
+		ESCAPE_REPLACEMENTS.put((int) '\\', "\\");
+		ESCAPE_REPLACEMENTS.put((int) Unicode.NEXT_LINE, "N");
+		ESCAPE_REPLACEMENTS.put((int) Unicode.NO_BREAK_SPACE, "_");
```
Comment (NathanSweet) [inline src/com/esotericsoftware/yamlbeans/emitter/EmitterWriter.java]: These constants do make sense.

**Comment by 0x1306e6d** [comment]
I'm sorry about wrong formatting and coding style.

This PR does not fix something but just extract something so I'll close this PR.
But how about Unicode constants? Are they good?

**Comment by NathanSweet** [comment]
Yes, the Unicode constants are good and make reading the emitter code nicer.

**Pull request closed without merging**